src/TrueOrFalse.Frontend.Web/Views/ApiVueApp/Stores/TopicToPrivateStoreController.cs

```diff
@@ -64,9 +64,10 @@ public JsonResult Get([FromRoute] int id)
 
             foreach (var c in aggregatedTopics)
             {
-                bool childHasPublicParent = c.Value.ParentCategories()
-                    .Any(p => p.Visibility == CategoryVisibility.All && p.Id != id);
-                if (!childHasPublicParent)
+                var parentCategories = c.Value.ParentCategories();
+                bool childHasPublicParent = parentCategories.Any(p => p.Visibility == CategoryVisibility.All && p.Id != id);
+
+                if (!childHasPublicParent && parentCategories.Any(p => p.Id != id))
                     return Json(new RequestResult
```

Comment:
@tj-mai This kind of logic should not be part of the controller logic. In future iterations, please create unit tests.